android/app/src/main/java/app/organicmaps/routing/NavigationService.java

```diff
@@ -210,16 +210,21 @@ public void onLocationUpdated(@NonNull Location location)
     if (!routingController.isNavigating())
       return;
 
+    final String[] turnNotifications = Framework.nativeGenerateNotifications();
+    if (turnNotifications != null)
+      TtsPlayer.INSTANCE.playTurnNotifications(getApplicationContext(), turnNotifications);
```

Comment:
Does it mean that turn notifications will be played on each location update now? Why it can't be done in the if below as before?

Reply:
No changes in the implementation, just moved from MwmActivity into this service:

https://github.com/organicmaps/organicmaps/blob/aa23f8e258dfb03ab3801ed563249fb5a7ca596b/android/src/app/organicmaps/MwmActivity.java#L1623-L1643

TtsPlayer.INSTANCE.playTurnNotifications() did `String[] turnNotifications = Framework.nativeGenerateNotifications();  if (turnNotifications != null) play()`.

Reply:
I am not sure that I understood your question fully. Please eleborate.

Reply:
You said that you moved code from MwmActivity, but it is not modified in PR.